examples/bsoncxx/builder_list.cpp

```diff
@@ -47,7 +47,7 @@ int main(int, char**) {
     //   "basic string" : "with value"
     // }
     //
-    doc = {"c-style", "with value", std::string("basic string"), "with value"};
+    doc = {{"c-style", "with value"}, {std::string("basic string"), "with value"}};
```

Comment:
Oh interesting, so it needs each key-value pair to be enclosed in curly braces?  It would probably feel more natural if we could avoid that and make it work with just commas, like in your previous iteration of this

Reply:
I agree that compile-time errors are preferable here, that's a strong argument for requiring std::pairs, even though it's a little more typing.